specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-07-01-preview/virtualNetworks.json

```diff
@@ -0,0 +1,575 @@
+          },
+          "description": "Subnet - list of subnets under the virtual network",
+          "x-ms-identifiers": []
+        },
+        "provisioningState": {
+          "type": "string"
+        },
+        "resourceName": {
+          "type": "string",
+          "description": "name of the object to be used in moc"
+        },
+        "status": {
+          "type": "object",
+          "description": "MOCVirtualNetworkStatus defines the observed state of MOCVirtualNetwork"
+        }
+      },
+      "description": "MOCVirtualNetworkSpec defines the desired state of MOCVirtualNetwork"
+    },
+    "IPPoolInfo": {
+      "type": "object",
+      "properties": {
+        "used": {
+          "type": "string",
+          "readOnly": true,
+          "description": "no. of ip addresses allocated from the ip pool"
+        },
+        "available": {
+          "type": "string",
+          "readOnly": true,
+          "description": "no. of ip addresses available in the ip pool"
+        }
+      }
+    },
+    "IPPool": {
+      "type": "object",
+      "properties": {
+        "ipPoolType": {
+          "type": "string",
+          "enum": [
+            "vm",
+            "vippool"
+          ],
+          "x-ms-enum": {
+            "name": "IPPoolTypeEnum",
+            "modelAsString": false
+          },
+          "description": "ip pool type"
+        },
+        "start": {
+          "type": "string",
+          "description": "start of the ip address pool"
+        },
+        "end": {
+          "type": "string",
+          "description": "end of the ip address pool"
+        },
+        "info": {
+          "$ref": "#/definitions/IPPoolInfo"
+        }
+      }
+    },
+    "virtualnetworks": {
+      "type": "object",
+      "x-ms-azure-resource": true,
+      "x-ms-client-flatten": true,
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/virtualnetworksProperties"
+        },
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Resource Id"
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Resource Name"
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Resource Type"
+        },
+        "location": {
+          "type": "string",
+          "description": "The resource location",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
+        },
+        "extendedLocation": {
```

Comment:
>extendedLocation [](http://example.com/codeflow?start=9&length=16)

Move to property bag.

Reply:
cc @JocelynBerrendonner

Reply:
We should talk to the custom location folks. They should add this to the property bag for every consumer of this code.

Reply:
Actually, let's first add it to a property bag we create. Other teams seem to have done that.